### PR TITLE
Update GNOME runtime

### PR DESCRIPTION
**Pull request by cho2**

### CODE REVIEW

**Comment by cho2** [comment]
bot, build

**Comment by AsciiWolf** [comment]
bot, build

**Comment by AsciiWolf** [comment]
```
{
    "errors": [
        "module-wxWidgets-autotools-non-release-build"
    ],
    "warnings": [
        "flathub-json-deprecated-i386-arch-included",
        "flathub-json-redundant-only-arches"
    ]
}
```

@cho2 Remove `i386` from flathub.json. I am not sure what exactly is wrong about the wxWidgets build, maybe the `--enable-debug` config-opt?

**Comment by cho2** [comment]
```
{
    "errors": [
        "module-wxWidgets-autotools-non-release-build"
    ],
    "warnings": [
        "flathub-json-redundant-only-arches"
    ]
}
```

**Comment by AsciiWolf** [comment]
Try updating Boost to a more recent version.

**Comment by AsciiWolf** [comment]
Also update WxWidgets to [3.1.7](https://github.com/wxWidgets/wxWidgets/releases/tag/v3.1.7) and bundle webkitgtk (that is probably missing from the 44 runtime for some reason):

```
checking for WEBKIT... configure: WARNING: webkit2gtk not found, falling back to webkitgtk
checking for WEBKIT... configure: WARNING: webkitgtk not found.
configure: WARNING: WebKit not available, disabling wxWebView
```

That will fix hopefully the last remaining issue.

[Here](https://github.com/flathub/org.gimp.GIMP/blob/master/org.gimp.GIMP.json#L191) is an example how to bundle it (although it probably will have to be slightly modified to work with Poedit).

File: flathub.json
Comment (hfiguiere) [inline flathub.json:1]: you can just remove this file

File: net.poedit.Poedit.json [hunk 1]
```diff
@@ -33,15 +33,14 @@
...
                 "--with-cxx=14",
                 "--enable-stl",
                 "--enable-webviewwebkit"
             ],
             "sources": [{
                     "type": "archive",
-                    "url": "https://github.com/wxWidgets/wxWidgets/releases/download/v3.1.4/wxWidgets-3.1.4.tar.bz2",
-                    "sha256": "3ca3a19a14b407d0cdda507a7930c2e84ae1c8e74f946e0144d2fa7d881f1a94"
+                    "url": "https://github.com/wxWidgets/wxWidgets/releases/download/v3.1.7/wxWidgets-3.1.7.tar.bz2",
+                    "sha256": "3d666e47d86192f085c84089b850c90db7a73a5d26b684b617298d89dce84f19"
```
Comment (hfiguiere) [inline net.poedit.Poedit.json:43]: why updating? Since it's the build problem

File: net.poedit.Poedit.json [hunk 2]
```diff
@@ -51,6 +50,49 @@
                     ]
                 }
             ]
+        },
+                {
+            "name": "webkitgtk",
+            "config-opts": [
+                "--with-gtk=2.0",
```
Comment (hfiguiere) [inline net.poedit.Poedit.json:57]: also I don't get why you'd build webkit gtk for gtk-2.0 when wxwidget is built for gtk3. Also why do you need to build webkit? It's in the platform.

**Comment by hfiguiere** [comment]
> Also update WxWidgets to [3.1.7](https://github.com/wxWidgets/wxWidgets/releases/tag/v3.1.7) and bundle webkitgtk (that is probably missing from the 44 runtime for some reason):
> 
> ```
> checking for WEBKIT... configure: WARNING: webkit2gtk not found, falling back to webkitgtk
> checking for WEBKIT... configure: WARNING: webkitgtk not found.
> configure: WARNING: WebKit not available, disabling wxWebView
> ```
> 
> That will fix hopefully the last remaining issue.
> 
> [Here](https://github.com/flathub/org.gimp.GIMP/blob/master/org.gimp.GIMP.json#L191) is an example how to bundle it (although it probably will have to be slightly modified to work with Poedit).

wxWidget hasn't been ported to use WebKit2Gtk-4.1, with the the version that uses libsoup3. And the example of GIMP is incorrect since this as about Gtk2.

Usually changing the detection for WebKit2Gtk to 4.1 is enough, but not in this case where we need to port the libsoup dependent code.

**Comment by hfiguiere** [comment]
example of what needs patching in wxwidget:
```
(cd /run/build/wxWidgets/lib/; rm -f libwx_baseu_net-3.1.so libwx_baseu_net-3.1.so.4; ln -s libwx_baseu_net-3.1.so.4.0.0 libwx_baseu_net-3.1.so.4; ln -s libwx_baseu_net-3.1.so.4 libwx_baseu_net-3.1.so)
./src/gtk/webview_webkit2.cpp: In function ‘gboolean wxgtk_webview_webkit_load_failed(WebKitWebView*, WebKitLoadEvent, gchar*, GError*, wxWebViewWebKit*)’:
./src/gtk/webview_webkit2.cpp:133:18: error: ‘SOUP_STATUS_CANCELLED’ was not declared in this scope; did you mean ‘SOUP_STATUS_ACCEPTED’?
  133 |             case SOUP_STATUS_CANCELLED:
      |                  ^~~~~~~~~~~~~~~~~~~~~
      |                  SOUP_STATUS_ACCEPTED
./src/gtk/webview_webkit2.cpp:137:18: error: ‘SOUP_STATUS_CANT_RESOLVE’ was not declared in this scope; did you mean ‘SOUP_STATUS_CONTINUE’?
  137 |             case SOUP_STATUS_CANT_RESOLVE:
      |                  ^~~~~~~~~~~~~~~~~~~~~~~~
      |                  SOUP_STATUS_CONTINUE
./src/gtk/webview_webkit2.cpp:142:18: error: ‘SOUP_STATUS_CANT_RESOLVE_PROXY’ was not declared in this scope; did you mean ‘SOUP_STATUS_USE_PROXY’?
  142 |             case SOUP_STATUS_CANT_RESOLVE_PROXY:
      |                  ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
      |                  SOUP_STATUS_USE_PROXY
./src/gtk/webview_webkit2.cpp:143:18: error: ‘SOUP_STATUS_CANT_CONNECT’ was not declared in this scope; did you mean ‘SOUP_STATUS_NO_CONTENT’?
  143 |             case SOUP_STATUS_CANT_CONNECT:
      |                  ^~~~~~~~~~~~~~~~~~~~~~~~
      |                  SOUP_STATUS_NO_CONTENT
./src/gtk/webview_webkit2.cpp:144:18: error: ‘SOUP_STATUS_CANT_CONNECT_PROXY’ was not declared in this scope
  144 |             case SOUP_STATUS_CANT_CONNECT_PROXY:
      |                  ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
./src/gtk/webview_webkit2.cpp:145:18: error: ‘SOUP_STATUS_SSL_FAILED’ was not declared in this scope; did you mean ‘SOUP_STATUS_SEE_OTHER’?
  145 |             case SOUP_STATUS_SSL_FAILED:
      |                  ^~~~~~~~~~~~~~~~~~~~~~
      |                  SOUP_STATUS_SEE_OTHER
./src/gtk/webview_webkit2.cpp:146:18: error: ‘SOUP_STATUS_IO_ERROR’ was not declared in this scope; did you mean ‘SOUP_STATUS_USE_PROXY’?
  146 |             case SOUP_STATUS_IO_ERROR:
      |                  ^~~~~~~~~~~~~~~~~~~~
      |                  SOUP_STATUS_USE_PROXY
./src/gtk/webview_webkit2.cpp:150:18: error: ‘SOUP_STATUS_MALFORMED’ was not declared in this scope; did you mean ‘SOUP_STATUS_LOCKED’?
  150 |             case SOUP_STATUS_MALFORMED:
      |                  ^~~~~~~~~~~~~~~~~~~~~
      |                  SOUP_STATUS_LOCKED
make: *** [Makefile:35936: webviewdll_webview_webkit2.o] Error 1
make: *** Waiting for unfinished jobs....
```

**Comment by cho2** [comment]
```
/app/include/lucene++/Lucene.h:131:26: warning: ‘template<class _Arg, class _Result> struct std::unary_function’ is deprecated [-Wdeprecated-declarations]
  131 | struct luceneHash : std::unary_function<TYPE, std::size_t> {
      |                          ^~~~~~~~~~~~~~
In file included from /usr/include/c++/12.2.0/functional:49,
                 from tm/transmem.h:30,
                 from tm/transmem.cpp:26:
/usr/include/c++/12.2.0/bits/stl_function.h:117:12: note: declared here
  117 |     struct unary_function
      |            ^~~~~~~~~~~~~~
/app/include/lucene++/Lucene.h:138:30: warning: ‘template<class _Arg, class _Result> struct std::unary_function’ is deprecated [-Wdeprecated-declarations]
  138 | struct luceneWeakHash : std::unary_function<TYPE, std::size_t> {
      |                              ^~~~~~~~~~~~~~
/usr/include/c++/12.2.0/bits/stl_function.h:117:12: note: declared here
  117 |     struct unary_function
      |            ^~~~~~~~~~~~~~
In file included from progressinfo.h:30,
                 from pretranslate.cpp:31:
titleless_window.h:32:24: warning: type attributes ignored after type is already defined [-Wattributes]
   32 | class WXDLLIMPEXP_CORE wxButton;
      |                        ^~~~~~~~
In file included from progressinfo.h:30,
                 from edframe.cpp:77:
titleless_window.h:32:24: warning: type attributes ignored after type is already defined [-Wattributes]
   32 | class WXDLLIMPEXP_CORE wxButton;
      |                        ^~~~~~~~
make[1]: *** [Makefile:716: fileviewer.o] Error 1
make[1]: *** Waiting for unfinished jobs....
In file included from welcomescreen.h:29,
                 from edapp.cpp:98:
titleless_window.h:32:24: warning: type attributes ignored after type is already defined [-Wattributes]
   32 | class WXDLLIMPEXP_CORE wxButton;
      |                        ^~~~~~~~
make[1]: Leaving directory '/run/build/poedit/src'
make: *** [Makefile:485: all-recursive] Error 1
FB: host_command_exited_cb 1419060 512
```

**Comment by hfiguiere** [comment]
it build fine with boost 1.74. the latest is not always the best

**Comment by hfiguiere** [comment]
these are only warning. But the actuall error is about webkit2

**Comment by hfiguiere** [comment]
Two solution: port wxWidget to use webkit2gtk 4.1 or build webkit2gtk 4.0 AND libsoup-2.4 (the latter in is shared-modules).

**Pull request closed without merging**